server/datastore/mysql/hosts.go

```diff
@@ -651,6 +651,11 @@ func (ds *Datastore) EnrollHost(ctx context.Context, osqueryHostID, nodeKey stri
 				return backoff.Permanent(ctxerr.Errorf(ctx, "host identified by %s enrolling too often", osqueryHostID))
 			}
 			hostID = int64(host.ID)
+
+			if err := cleanupPolicyMembershipOnTeamChange(ctx, tx, teamID, []uint{host.ID}); err != nil {
+				return ctxerr.Wrap(ctx, err, "EnrollHost delete policy membership")
+			}
+
```

Comment:
I'm not sure if running this in `EnrollHost` is desirable, this is the most straightforward solution but enrolling hosts seems like a hot path.

Should this be part of a scheduled job?

Reply:
Hm, yeah, we should keep this as lean as possible. Doing things in a scheduled job will have the problem of the time it takes between the team change and the cleanup running.

How about we get the team_id when we do 

```
SELECT id, last_enrolled_at FROM hosts WHERE osquery_host_id = ?
```

And if it's different from the one we get as a param, then we cleanup, otherwise we skip it?

Reply:
that sounds great 👍